src/hotspot/share/classfile/javaAssertions.cpp

```diff
@@ -135,8 +136,8 @@ objArrayHandle names, typeArrayHandle enabled, TRAPS) {
   int index;
   for (index = len - 1; p != 0; p = p->next(), --index) {
     assert(index >= 0, "length does not match list");
-    Handle s = java_lang_String::create_from_str(p->name(), CHECK);
-    s = java_lang_String::char_converter(s, JVM_SIGNATURE_SLASH, JVM_SIGNATURE_DOT, CHECK);
+    TempNewSymbol name = SymbolTable::new_symbol(p->name());
```

Comment:
Why do we need a temp symbol?

Reply:
Because the new_symbol call will increment the Symbol* refcount in the SymbolTable, it's then copied to the java.lang.String, so the name isn't needed after this.

Reply:
Ah I see. The types were confusing me. We start with a char* convert to a Symbol, then convert that back to char* but with the externalized format. The irony there being that  it seems these Options start in external format but then get stored in internal format!
No issue.

Reply:
Yes, Ioi and I discussed this part of the change but since we substitute slashes for dots and then back again, it is correct.  Thanks!